File: guess.rb
number = rand(1..20)

puts 'Hello! What is your name?'
name = gets&.chomp

puts "Hi, #{name}. I'm thinking of a number between 1 and 20."

1.upto 6 do |guesses|
  puts "What is your guess? You have #{7 - guesses} more guesses."
  guess = gets&.chomp.to_i

  if guess == number
    puts "Good job, #{name}! You guessed my number in #{guesses} guesses."
    exit
  else
    puts(guess > number ? 'Too high' : 'Too low')
  end
end

puts "Nope. The number I was thinking of was #{number}."

File: guess.py
import random


number = random.randint(1, 20)
guesses = 0

print('Hello! What is your name?')
name = input()

print(f"Hi, {name}. I'm thinking of a number from 1 and 20.")

while guesses < 6:
    print(f'What is your guess? You have {6 - guesses} more guesses.')
    guess = input()
    guess = int(guess)

    guesses = guesses + 1

    if guess < number:
        print('Too low.')
    elif guess > number:
        print('Too high.')
    elif guess == number:
        print(f'Good job, {name}! You guessed my number in {guesses} guesses!')
        break

if guess != number:
    print(f'Nope. The number I was thinking of was {number}.')
